r/asg: AMI that listens on port 80, Nginx stack or other?

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

I think a good approach would be to stand up a base instance and install Nginx stack in userdata. 

Fixed = `TestAccAWSAutoScalingGroup_WithLoadBalancer_ToTargetGroup` and `TestAccAWSAutoScalingGroup_WithLoadBalancer` run successfully on GovCloud.

### Terraform CLI and Terraform AWS Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

```
Terraform v0.12.29
AWS v3.0.0
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_autoscaling_group

### Terraform Configuration Files

NO HCL CHANGES

### Acceptance Test Output

On commercial, the AMI is found with `bitnami-nginxstack-*-linux-debian-9-x86_64-hvm-ebs` but, the same query on GovCloud yields, `Your query returned no results`.

Problem config:

```hcl
// need an AMI that listens on :80 at boot, this is:
data "aws_ami" "test_ami" {
  most_recent = true

  owners = ["979382823631"]

  filter {
    name   = "name"
    values = ["bitnami-nginxstack-*-linux-debian-9-x86_64-hvm-ebs"]
  }
}
```

Acc testing output:
```
TestAccAWSAutoScalingGroup_WithLoadBalancer_ToTargetGroup: testing.go:684: Step 0 error: errors during refresh:
        
        Error: Your query returned no results. Please change your search criteria and try again.
        
          on /var/folders/g2/jg8wrkmn21xcpcx_00d78rtc0000gp/T/tf-test999368337/main.tf line 73:
          (source code not available)
        
--- FAIL: TestAccAWSAutoScalingGroup_WithLoadBalancer_ToTargetGroup (3.11s)
    
TestAccAWSAutoScalingGroup_WithLoadBalancer: testing.go:684: Step 0 error: errors during refresh:
        
        Error: Your query returned no results. Please change your search criteria and try again.
        
          on /var/folders/g2/jg8wrkmn21xcpcx_00d78rtc0000gp/T/tf-test087845611/main.tf line 73:
          (source code not available)
        
--- FAIL: TestAccAWSAutoScalingGroup_WithLoadBalancer (3.18s)
```


### Expected Behavior

<!--- What should have happened? --->
Acceptance tests succeed on _GovCloud_.

### Actual Behavior

<!--- What actually happened? --->
Acceptance tests fail on _GovCloud_.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

Either:

1. `TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run= TestAccAWSAutoScalingGroup_WithLoadBalancer -timeout 120m`
1. `TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run= TestAccAWSAutoScalingGroup_WithLoadBalancer_ToTargetGroup -timeout 120m`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #14222

--- a/aws/resource_aws_autoscaling_group_test.go
+++ b/aws/resource_aws_autoscaling_group_test.go
@@ -2315,8 +2315,10 @@ resource "aws_autoscaling_group" "bar" {
 }
 
 func testAccAWSAutoScalingGroupConfigWithLoadBalancer() string {
-	return testAccAvailableAZsNoOptInDefaultExcludeConfig() +
-		fmt.Sprintf(`
+	return composeConfig(
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableAZsNoOptInDefaultExcludeConfig(),
+		`
 resource "aws_vpc" "foo" {
   cidr_block = "10.1.0.0/16"
   tags = {
@@ -2377,22 +2379,17 @@ resource "aws_elb" "bar" {
   depends_on = ["aws_internet_gateway.gw"]
 }
 
-// need an AMI that listens on :80 at boot, this is:
-data "aws_ami" "test_ami" {
-  most_recent = true
-
-  owners = ["979382823631"]
-
-  filter {
-    name   = "name"
-    values = ["bitnami-nginxstack-*-linux-debian-9-x86_64-hvm-ebs"]
-  }
-}
-
 resource "aws_launch_configuration" "foobar" {
-  image_id        = data.aws_ami.test_ami.id
+  image_id        = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
   instance_type   = "t2.micro"
   security_groups = [aws_security_group.foo.id]
+
+  # Need the instance to listen on port 80 at boot
+  user_data = <<EOF
+#!/bin/bash
+echo "Terraform aws_autoscaling_group Testing" > index.html
+nohup python -m SimpleHTTPServer 80 &
+EOF
 }
 
 resource "aws_autoscaling_group" "bar" {
@@ -2411,8 +2408,10 @@ resource "aws_autoscaling_group" "bar" {
 }
 
 func testAccAWSAutoScalingGroupConfigWithTargetGroup() string {
-	return testAccAvailableAZsNoOptInDefaultExcludeConfig() +
-		fmt.Sprintf(`
+	return composeConfig(
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableAZsNoOptInDefaultExcludeConfig(),
+		`
 resource "aws_vpc" "foo" {
   cidr_block = "10.1.0.0/16"
   tags = {
@@ -2479,22 +2478,17 @@ resource "aws_elb" "bar" {
   depends_on = ["aws_internet_gateway.gw"]
 }
 
-// need an AMI that listens on :80 at boot, this is:
-data "aws_ami" "test_ami" {
-  most_recent = true
-
-  owners = ["979382823631"]
-
-  filter {
-    name   = "name"
-    values = ["bitnami-nginxstack-*-linux-debian-9-x86_64-hvm-ebs"]
-  }
-}
-
 resource "aws_launch_configuration" "foobar" {
-  image_id        = data.aws_ami.test_ami.id
+  image_id        = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
   instance_type   = "t2.micro"
   security_groups = [aws_security_group.foo.id]
+
+  # Need the instance to listen on port 80 at boot
+  user_data = <<EOF
+#!/bin/bash
+echo "Terraform aws_autoscaling_group Testing" > index.html
+nohup python -m SimpleHTTPServer 80 &
+EOF
 }
 
 resource "aws_autoscaling_group" "bar" {

